backend/src/services/release.ts

```diff
@@ -295,6 +326,168 @@ export async function getReleaseBySemver(user: UserInterface, modelId: string, s
   return release
 }
 
+function getSemverQueryBounds(querySemver: string) {
+  const semverRangeStandardised = semver.validRange(querySemver, { includePrerelease: false })
+
+  if (!semverRangeStandardised) {
+    throw BadReq(`Semver range, '${querySemver}' is invalid `)
+  }
+
+  const [expressionA, expressionB] = semverRangeStandardised.split(' ')
+
+  let lowerSemver: string = '',
+    upperSemver: string = '',
+    lowerInclusivity = false,
+    upperInclusivity = false
+  if (expressionA.includes('>')) {
```

Comment:
When an if statement becomes this complex it's always worth seeing if it could be simplified.
Have you considered:
```
const lowerInclusivity = expressionA.includes('>=')
const upperInclusivity = expressionB.includes('<=') | expressionA.includes('<=')
const lowerSemver = expressionA.replace(/[<>=]/g, '')
const upperSemver = expressionB.replace(/[<>=]/g, '')

Reply:
Yeah, regex makes this a tonne easier, good idea

Reply:
Although, expressionA can be upperSemver, not just lowerSemver. I will a check to see whether upperSemver will be expressionA or B.

Reply:
Done, still required some if statements as expressionA can be the upperSemver as the lower/upper parts of the variables state wether they are upper or lower bounds, not first or second in query